packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -257,11 +265,6 @@ export class PolicyStatement extends Token {
     return this.resource && this.resource.length > 0;
   }
 
-  public describe(sid: string): PolicyStatement {
```

Comment:
I hate the name of this method, but why did you decide that this must be passed in the ctor? Why not just rename this to `addSid(sid)`

Reply:
I kinda wanted it to be immutable... There's a bunch of places that need to guarantee uniqueness (and presence) of SIDs (like SNS Topic policies), and it felt wrong to not be able to offer this guarantee safely at construction time.

Reply:
How about just protecting against double-setting the SID. I think that's what you are trying to prevent. You can see the downside of this change in the SNS code, where now users have to artificially supply a SID, whereas we can do that for them quite safely.

Reply:
Yeah - well the alternative approach I mentioned is actually *much much* nicer and thus I won't touch this at all anymore... Yay.